### ISSUE TITLE
Help me understand documentation copyright for the messaging spec

### ISSUE THREAD

**Issue by gibiansky** [jupyter/jupyter_client#166]
I am creating a Haskell library called `jupyter-kernel` (or maybe `jupyter`) to implement the messaging spec and replace the base layer of IHaskell. When documenting the data types for all the message types and fields, it would make the *most* sense for me to just copy parts of the documentation that you have in the messaging spec into the code, as that is the most true to what *should* be in the library documentation (since the library data types should mirror precisely in a predictable way what the messaging spec says). Of course, I *can* try to reword it to avoid directly copy-pasting, but that doesn't seem like a productive use of time and seems like it may even introduce errors.

Is this an acceptable use or fair use of the messaging spec documentation?

I don't know what the right thing to do here is, so I figured the best thing would be to just ask you directly.

**Comment by Carreau** [comment]
> Is this an acceptable use or fair use of the messaging spec documentation?

I'll ask the steering council, but I think it's ok to reuse as is.

**Comment by Carreau** [comment]
I've sent a request for 1) ask if it's or for you in this case to use it as-is, 2) Make it clearer in general. 

While waiting for the response, is there anything we can do that would make your usage of it easier, like to avoid divergence of the 2 documentations ? And/or avoid you to copy-past things ?

**Comment by gibiansky** [comment]
Thanks @Carreau 

I would love for their to be away to autogenerate some parts of this or avoid this duplication but I think it's not really doable or worth the effort given the size of the messaging spec. I will try to include many links to the official docs so that it's clear where info comes from, and in general probably the best thing would be to just have very thorough changelogs during releases (in addition to bullet points of all changed messaging spec items)

**Comment by Carreau** [comment]
> I would love for their to be away to autogenerate some parts of this or avoid this duplication but I think it's not really doable or worth the effort given the size of the messaging spec. I will try to include many links to the official docs so that it's clear where info comes from, and in general probably the best thing would be to just have very thorough changelogs during releases (in addition to bullet points of all changed messaging spec items)

Great. 

The current status of the discussion is:
 - At least it's BSD, so Copy is OK as long as there is a note of the original license in your version. 
 - It seem we don't have issues with having Documentation/Example as CC0, but will all the contributors it might be hard to relicense.

**Comment by gibiansky** [comment]
The project is MIT licensed already, sadly...

On Tue, Jul 5, 2016, 8:20 AM Matthias Bussonnier <notifications@github.com>
wrote:

> I would love for their to be away to autogenerate some parts of this or
> avoid this duplication but I think it's not really doable or worth the
> effort given the size of the messaging spec. I will try to include many
> links to the official docs so that it's clear where info comes from, and in
> general probably the best thing would be to just have very thorough
> changelogs during releases (in addition to bullet points of all changed
> messaging spec items)
>
> Great.
>
> The current status of the discussion is:
>
>    - At least it's BSD, so Copy is OK as long as there is a note of the
>    original license in your version.
>    - It seem we don't have issues with having Documentation/Example as
>    CC0, but will all the contributors it might be hard to relicense.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/jupyter/jupyter_client/issues/166#issuecomment-230510360>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABx2w6ZV1s_HuNE0DJh2BTZQAyaAvaiwks5qSnZLgaJpZM4JErDQ>
> .
>
-- 

– Andrew

**Comment by takluyver** [comment]
There should be no problem with including a BSD-licensed file in an MIT licensed project - just stick the BSD notice somewhere into that file.
